frontend/src/features/leases/add/tenant/AddLeaseTenantNote.tsx

```diff
@@ -0,0 +1,56 @@
+import { TextArea } from 'components/common/form';
+import GenericModal from 'components/common/GenericModal';
+import { IconButton } from 'components/common/styles';
+import { getIn, useFormikContext } from 'formik';
+import useDeepCompareEffect from 'hooks/useDeepCompareEffect';
+import * as React from 'react';
+import { useState } from 'react';
+import { FaRegFileAlt } from 'react-icons/fa';
+import { withNameSpace } from 'utils/formUtils';
+
+interface IAddLeaseTenantNoteProps {
+  nameSpace?: string;
+}
+
+const AddLeaseTenantNote: React.FunctionComponent<IAddLeaseTenantNoteProps> = ({ nameSpace }) => {
+  const { values, setFieldValue } = useFormikContext();
+  const [showNotes, setShowNotes] = useState(false);
+  const [currentNote, setCurrentNote] = useState();
+  const field = withNameSpace(nameSpace, 'note');
+  const noteValue = getIn(values, field);
+  const summaryValue = getIn(values, withNameSpace(nameSpace, 'summary'));
+
+  useDeepCompareEffect(() => {
+    if (showNotes === true) {
+      setCurrentNote(noteValue);
+    }
+  }, [showNotes]);
```

Comment:
question - why do we need **useDeepCompareEffect** here if `showNotes` is boolean? Wouldn't it be equivalent to `useEffect` ?

Reply:
hmm, I think I iterated on this a few times, useDeepCompareEffect was required previously but should not be any more.